batch.py

```diff
@@ -309,6 +319,24 @@ def crm_pipeline(
     else:
         genes_of_interest = list(gene_dict.keys())
 
+    # Setup MAX concurrency by genes
+    _prev_dependent_jobs: list[hb.job.Job] = []
+    _new_dependent_jobs: list[hb.job.Job] = []
+
+    def manage_concurrency_for_job(job: hb.job.Job):
```

Comment:
A maybe simpler implementation of this is to depend on `jobs[my_index - max_concurrency]`.

Reply:
Yeah much better idea, I've actioned this, could you review again?